stpctl/stpctl.c

```diff
@@ -0,0 +1,459 @@
+                    msg.dbg.event = 1;
+                else if (0 == strncmp("off", argv[3], 3))
+                    msg.dbg.event = 0;
+            }
+            else if (0 == strncmp("port", argv[2], strlen("port")))
+            {
+                msg.dbg.flags |= STPCTL_DBG_SET_PORT;
+                if (0 == strncmp("all", argv[3], 3))
+                    msg.intf_name[0] = '\0';
+                else
+                    strncpy(msg.intf_name, argv[3], IFNAMSIZ);
+                if (0 == strncmp("on", argv[4], 2))
+                    msg.dbg.port = 1;
+                else if (0 == strncmp("off", argv[4], 3))
+                    msg.dbg.port = 0;
+            }
+            else if (0 == strncmp("vlan", argv[2], strlen("vlan")))
+            {
+                msg.dbg.flags |= STPCTL_DBG_SET_VLAN;
+                if (0 == strncmp("all", argv[3], 3))
+                    msg.vlan_id = 0;
+                else
+                    msg.vlan_id = strtol(argv[3], &end_ptr, 10);
+                if (0 == strncmp("on", argv[4], 2))
+                    msg.dbg.vlan = 1;
+                else if (0 == strncmp("off", argv[4], 3))
+                    msg.dbg.vlan = 0;
+            }
+            else if (0 == strncmp("show", argv[2], strlen("show")))
+            {
+                msg.dbg.flags |= STPCTL_DBG_SHOW;
+            }
+            else
+            {
+                stpout("invalid argv[2] : %s\n", argv[2]);
+                return -1;
+            }
+
+            break;
+        }
+
+        case STP_CTL_DUMP_LIBEV_STATS:
+        {
+            /* No arg */
+            if (!(argc == 2))
+            {
+                stpout("invalid number of args\n");
+                return -1;
+            }
+            break;
+        }
+
+        case STP_CTL_CLEAR_ALL:
+        {
+            /* No arg */
+            if (!(argc == 2))
+            {
+                stpout("invalid number of args\n");
+                return -1;
+            }
+            break;
+        }
+
+        case STP_CTL_CLEAR_VLAN:
+        {
+            if (!(argc == 3))
+            {
+                stpout("invalid number of args\n");
+                return -1;
+            }
+
+            msg.vlan_id = atoi(argv[2]);
+            break;
+        }
+
+        case STP_CTL_CLEAR_INTF:
+        {
+            if (!(argc == 3))
+            {
+                stpout("invalid number of args\n");
+                return -1;
+            }
+
+            strncpy(msg.intf_name, argv[2], IFNAMSIZ);
+            break;
+        }
+
+        case STP_CTL_CLEAR_VLAN_INTF:
+        {
+            if (!(argc == 4))
+            {
+                stpout("invalid number of args\n");
+                return -1;
+            }
+
+            msg.vlan_id = atoi(argv[2]);
+            strncpy(msg.intf_name, argv[3], IFNAMSIZ);
+            break;
+        }
+
+        default:
+        {
+            stpout("invalid command %d\n", cmd_type);
```

Comment:
send_command() calling place the socket gets closed and program exists, do we need this explicitly here?

Reply:
Yes, removed